File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy=prices[0];
        int profit=0;
        int mprofit=Integer.MIN_VALUE;
        for(int i=1;i<prices.length;i++){
            if(prices[i]<buy){
                buy=prices[i];
            }
            if(prices[i]>buy){
                profit=prices[i]-buy;
            }
            if(profit>mprofit){
                mprofit=profit;
            }
        }
       if(mprofit==Integer.MIN_VALUE)
            return 0;
        else
            return mprofit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        buy=prices[0]
        profit=0
        mprofit=0
        for i in range(1,len(prices)):
            if(prices[i]<buy):
                buy=prices[i]
            if(prices[i]>buy):
                profit=prices[i]-buy
            if(profit>mprofit):
                mprofit=profit
        return mprofit            